src/generators/schema/author.php

```diff
@@ -112,7 +112,12 @@ public function generate( Meta_Tags_Context $context ) {
 	 * @return bool|int User ID or false upon return.
 	 */
 	protected function determine_user_id( Meta_Tags_Context $context ) {
-		$user_id = (int) $context->post->post_author;
+		$user_id = 0;
```

Comment:
I think this should either default to `false` or the method's return value description should be adapted to say that `0` means that no applicable user can be found.

Reply:
Good point. Though there are some complications with that:
- The `(int) $context->post->post_author;` could very well have `0` when no post_author is set.
- We return the filtered value so we don't actually enforce any type.

So a check after the filter seems wise at this point? 🤔 Which would then also need to be added to the Author. The tests would need to be adapted too, which we are currently working on.

I think it's better to not fix that in this PR. I probably should not have changed this to begin with... should've made an issue instead. Mudding the scope. I have the code now, but it needs test instructions too and if not clear it will make the rest of this halt too. Making a separate issue.

Reply:
https://github.com/Yoast/wordpress-seo/issues/14615